packages/sanity/src/core/studio/components/navbar/search/contexts/search/SearchProvider.tsx

```diff
@@ -140,6 +142,10 @@ export function SearchProvider({children, fullscreen}: SearchProviderProps) {
           skipSortByScore: ordering.ignoreScore,
           ...(ordering.sort ? {sort: [ordering.sort]} : {}),
           cursor: cursor || undefined,
+          perspective: omitBundlePerspective(perspective),
+          bundlePerspective: perspective?.startsWith('bundle.')
+            ? [perspective.split('bundle.').at(1), DRAFTS_FOLDER].join(',')
+            : undefined,
```

Comment:
probably overkill, but it took me a bit to understand what was going on here.
Wdyt if we move this to a function that returns the params we need.

```ts
const getPerspective = (
  perspective: string | undefined,
):
  | {
      bundlePerspective: string
    }
  | {
      perspective: string | undefined
    } => {
  if (perspective?.startsWith('bundle.')) {
    const [_identifier, bundleId] = perspective.split('bundle.')
    // When querying bundles, we should use the bundleId and the drafts as the bundlePerspective e.g. [bundleId, 'drafts']
    return {bundlePerspective: [bundleId, DRAFTS_FOLDER].join(',')}
  }
  return {
    perspective: perspective,
  }
}

```

Reply:
We could do the same in the text search

Reply:
I think it's a great point. It's definitely not the most intuitive code to read.